tests/hashed_token_router.rs

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+use std::net::{Ipv6Addr, SocketAddr};
+
+use tokio::time::{timeout, Duration};
+
+use quilkin::{
+    config::Filter,
+    filters::{Capture, HashedTokenRouter, StaticFilter},
+    net::endpoint::{metadata::MetadataView, Endpoint},
+    test::{AddressType, TestHelper},
+};
+
+/// This test covers both hashed_token_router and capture filters,
+/// since they work in concert together.
+#[tokio::test]
+async fn hashed_token_router() {
+    let mut t = TestHelper::default();
+    let mut echo = t.run_echo_server(AddressType::Ipv6).await;
+    quilkin::test::map_to_localhost(&mut echo).await;
+
+    let capture_yaml = "
+suffix:
+    size: 3
+    remove: true
+";
+    let endpoint_metadata = "
+quilkin.dev:
+    tokens:
+        - YWJj # abc
+        ";
+
+    let server_config = std::sync::Arc::new(quilkin::Config::default_non_agent());
+    server_config.clusters.modify(|clusters| {
+        clusters.insert_default(
+            [Endpoint::with_metadata(
+                echo.clone(),
+                serde_yaml::from_str::<MetadataView<_>>(endpoint_metadata).unwrap(),
+            )]
+            .into(),
+        );
+
+        clusters.build_token_maps();
```

Comment:
Should this maybe happen through a watcher task in quilkin instead, to prevent a code path where the token maps aren't built from ever happening.

Reply:
Well it does create them automatically when endpoint sets are updated in a proxy, this function call is just exposed since tests/benches aren't doing the normal operation but I suppose I could change it so that they're either built like they are now, or else when accessed, the issue would be that if they're built on access in actual running code that would be expensive.

Reply:
Alright, I guess it's fine for now.